region/brinstar/green/Green Brinstar Main Shaft.json

```diff
@@ -3960,7 +3960,7 @@
         }
       },
       "requires": [
-        "h_EtecoonDoorSpawnFix"
+        "h_DoorImmediatelyClosedFix"
```

Comment:
Does the door spawn need to be fixed for it to block going back through?  Like it won't be there without a spawnFix patch?

Reply:
In vanilla, the door "closes" one screen to the left, as well as already being instantly closed. 

Apparently all doors are already closed and then immediately afterwards there is a PLM drawn over them, and the one you just came through is opened and then starts closing. 

That's why in indirect g-mode, all doors are closed. In direct, PLMs start processing a bit before the jank happens, so the door you came through is replaced with the open door PLM, and items and gates are added.

Reply:
In Etecoon shaft, the closing PLM was placed a screen to the left, and it was never caught because the door is permanently grey, so it didnt really get testing.

Apparently, each door has a bit that says whether or not a door should close, so for transitions without a door cap it is turned off. Pink PB room was probably messed up because the door on the right is Big Pink, and it has the bit turned off because there is no door there. 

Halfie Climb is pretty surprising though